source/technical/complex.rst

```diff
@@ -739,9 +739,52 @@ The ``SoftwareDeploymentGroup`` resources ``export_hosts_sdg`` and ``populate_ho
 
 Finally, the instance ``ResourceGroup`` is configured so that each instance uses the following contextualization method instead of a ``user_data`` script:
 
-.. code:: 
+.. code::
 
    user_data_format: SOFTWARE_CONFIG
    software_config_transport: POLL_SERVER_HEAT
 
 You can follow the same template pattern to configure your own deployment requiring all-to-all information exchange.
+
+.. _advanced-reservation-orchestration:
+
+Advanced Reservation Orchestration
+_______________________________
+
+On Chameleon you can configure a Heat Stack to launch as soon as your lease begins. Whether your experiments require a large cluster or a single node, orchestrating an advanced reservation is can save you time configuring your environment or provide a blueprint for your experiment that will run automatically
+when the necessary resources become available.
+
+At present, you will need to use our customized versions the Heat and Blazar CLI tools to implement this feature. We are currently working to provide support for this functionality through the GUI.
+
+Install Custom CLI
+~~~~~~~~~~~
+
+Clone the Chameleon repositories for ``python-heatclient`` and ``python-blazarclient``. You can then simply run ``python setup.py install`` in the repo folder to override the your existing python client packages if they are already installed. Links and sample code below:
+
+.. code::
+
+    git clone https://github.com/ChameleonCloud/python-heatclient.git
+    python /path/to/python-heatclient_repo/setup.py install
+
+    git clone https://github.com/ChameleonCloud/python-blazarclient.git
+    python /path/to/python-blazarclient_repo/setup.py install
+
+
+Initialize Stack
+~~~~~~~~~~~
+
+Next you will need to configure a Heat stack with the ``--initialize`` flag on the CLI and a dummy ``reservation_id`` parameter. This will validate your template and store all the data required to launch a stack without Heat attempting to allocate resources. See example command below:
+
+.. code::
+
+    openstack stack create -t <template_file> --initialize --parameter reservation_id=dummy <stack_name>
```

Comment:
Is this `dummy` value important? Maybe call out that it can be any value? Also, is it possible to just leave it blank?

Reply:
It can be an empty string, but you can't leave out the parameter altogether without getting a 'missing parameter' error. The exception being if you provide a default value for a parameter in the Heat Template.